ui/pages/Project/selectors.js

```diff
@@ -207,33 +226,49 @@ export const getFamiliesSortDirection = createSelector(
   familyTableState => familyTableState.familiesSortDirection || 1,
 )
 
-/**
- * function that returns an array of family guids that pass the currently-selected
- * familiesFilter.
- *
- * @param state {object} global Redux state
- */
+const hasFamilySearch = createSelector(
+  getFamiliesSearch,
+  familiesSearch => !!familiesSearch,
+)
+
+const getFamilySearchFields = (family, individualsByGuid) => ([
+  family.displayName, family.familyId, (family.assignedAnalyst || {}).fullName, (family.assignedAnalyst || {}).email,
+  ...family.analysedBy.map(({ createdBy }) => `${createdBy.fullName}${createdBy.email}`),
+  ...family.individualGuids.map(individualGuid => (individualsByGuid[individualGuid].features || []).map(feature => feature.label).join(';')),
+])
+
+const getFamiliesBySearchString = createSelector(
+  getProjectAnalysisGroupFamiliesByGuid,
+  getIndividualsByGuid,
+  hasFamilySearch,
+  (familiesByGuid, individualsByGuid, shouldSearch) => {
+    if (!shouldSearch) {
+      return null
+    }
+
+    return Object.values(familiesByGuid).reduce((acc, family) => (
+      { ...acc, [getFamilySearchFields(family, individualsByGuid).join(';').toLowerCase()]: family }), {})
+  },
+)
+
 export const getVisibleFamilies = createSelector(
   getProjectAnalysisGroupFamiliesByGuid,
+  getFamiliesBySearchString,
   getIndividualsByGuid,
-  getSamplesByGuid,
+  getSamplesByFamily,
   getUser,
   getFamiliesFilter,
   getFamiliesSearch,
-  (familiesByGuid, individualsByGuid, samplesByGuid, user, familiesFilter, familiesSearch) => {
-    const searchFilter = familiesSearch ? family =>
```

Comment:
before, whenever anyone would type a new character into the filter bar we would re-compute the search string for each family and check if it matched the new filter. Now, we have one selector that will compute all the search strings if anything has been searched, and then we reuse that computation whenever the filter value changes to check for the new search string in them. This lets us reuse a lot of computations, which is good